copilot-core/src/Copilot/Core/Spec.hs

```diff
@@ -29,17 +29,19 @@ import Data.Typeable (Typeable)
 -- Internal imports
 import Copilot.Core.Expr (Expr, Id, Name, UExpr)
 import Copilot.Core.Type (Type, Typed)
+import GHC.Stack (CallStack)
 
 -- | A stream in an infinite succession of values of the same type.
 --
 -- Stream can carry different types of data. Boolean streams play a special
 -- role: they are used by other parts (e.g., 'Trigger') to detect when the
 -- properties being monitored are violated.
 data Stream = forall a . (Typeable a, Typed a) => Stream
-  { streamId       :: Id
-  , streamBuffer   :: [a]
-  , streamExpr     :: Expr a
-  , streamExprType :: Type a
+  { streamId        :: Id
+  , streamBuffer    :: [a]
+  , streamExpr      :: Expr a
+  , streamExprType  :: Type a
+  , streamCallStack :: CallStack
```

Comment:
It might be worth leaving a comment in this file somewhere explaining what the goal of adding `CallStack` is: namely, to ensure that we can obtain the line number of the place in the specification where each function is called at the bottom of the call stack. (The rest of the call stack isn't terribly important, so we can leave the rest unspecified.)